components/http/okHttp/src/main/java/com/microsoft/kiota/http/OkHttpRequestAdapter.java

```diff
@@ -642,45 +566,35 @@ private Response throwIfFailedResponse(@Nonnull final Response response, @Nonnul
+        Objects.requireNonNull(requestInfo, nullRequestInfoParameter);
         final Span span = GlobalOpenTelemetry.getTracer(obsOptions.getTracerInstrumentationName()).spanBuilder("getHttpResponseMessage").setParent(Context.current().with(parentSpan)).startSpan();
         try(final Scope scope = span.makeCurrent()) {
             this.setBaseUrlForRequestInformation(requestInfo);
-            final Map<String, Object> additionalContext = new HashMap<String, Object>() {{
-                put("parent-span", span);
-            }};
+            final Map<String, Object> additionalContext = new HashMap<String, Object>();
+            additionalContext.put("parent-span", span);
             if(claims != null && !claims.isEmpty()) {
                 additionalContext.put(claimsKey, claims);
             }
-            return this.authProvider.authenticateRequest(requestInfo, additionalContext)
-            .thenCompose(x -> {
-                try {
-                    final OkHttpCallbackFutureWrapper wrapper = new OkHttpCallbackFutureWrapper();
-                    this.client.newCall(getRequestFromRequestInformation(requestInfo, span, spanForAttributes)).enqueue(wrapper);
-                    return wrapper.future;
-                } catch (URISyntaxException | MalformedURLException ex) {
-                    spanForAttributes.recordException(ex);
-                    final CompletableFuture<Response> result = new CompletableFuture<Response>();
-                    result.completeExceptionally(ex);
-                    return result;
-                }
-            })
-            .thenApply(x -> {
-                final String contentLengthHeaderValue = getHeaderValue(x, "Content-Length");
-                if(contentLengthHeaderValue != null && !contentLengthHeaderValue.isEmpty()) {
-                    final Integer contentLengthHeaderValueAsInt = Integer.parseInt(contentLengthHeaderValue);
-                    spanForAttributes.setAttribute(SemanticAttributes.HTTP_RESPONSE_CONTENT_LENGTH, contentLengthHeaderValueAsInt);
-                }
-                final String contentTypeHeaderValue = getHeaderValue(x, "Content-Length");
-                if(contentTypeHeaderValue != null && !contentTypeHeaderValue.isEmpty()) {
-                    spanForAttributes.setAttribute("http.response_content_type", contentTypeHeaderValue);
-                }
-                spanForAttributes.setAttribute(SemanticAttributes.HTTP_STATUS_CODE, x.code());
-                spanForAttributes.setAttribute(SemanticAttributes.HTTP_FLAVOR, x.protocol().toString().toUpperCase(Locale.ROOT));
-                return x;
-            })
-            .thenCompose(x -> this.retryCAEResponseIfRequired(x, requestInfo, span, spanForAttributes, claims));
+            this.authProvider.authenticateRequest(requestInfo, additionalContext);
+            this.client.newCall(getRequestFromRequestInformation(requestInfo, span, spanForAttributes));
```

Comment:
```suggestion
```
I'm not sure why we're making the call twice here?

Reply:
oversight I believe, will correct in the next commit.